pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1888,6 +1883,51 @@ func (service *SecurityPolicyService) buildRulePortsString(ports *[]v1alpha1.Sec
 	return util.GenerateTruncName(common.MaxNameLength, portsString, "", suffix, "", "")
 }
 
+func (service *SecurityPolicyService) buildRulePortNumberString(port *v1alpha1.SecurityPolicyPort, hasNamedport bool, portNumber int) string {
+	// Build the rule port number string name for non named port.
+	// This is a common case where the string is built from port definition. For instance,
+	// - protocol: TCP
+	//   port: 8282
+	//   endPort: 8286
+	// The built port number string is: 8282.8286
+	// - protocol: UDP
+	//   port: 3308
+	// The built port number string is: 3308
+	if !hasNamedport {
+		if port.EndPort != 0 {
+			return fmt.Sprintf("%s.%d", (port.Port).String(), port.EndPort)
+		}
+		return fmt.Sprintf("%s", (port.Port).String())
+	} else {
```

Comment:
Can we use this format?

```
func (service *SecurityPolicyService) buildRulePortNumberString(port v1alpha1.SecurityPolicyPort, portNumber int) string {
	if port.Port.Type == intstr.String {
		return fmt.Sprintf("%d", portNumber)
	}
	if port.EndPort != 0 {
		return fmt.Sprintf("%s.%d", (port.Port).String(), port.EndPort)
	}
	return (port.Port).String()
}
```

Reply:
Thanks for suggestion, refactor the code as suggested.